content/chat/getting-started/javascript.textile

```diff
@@ -0,0 +1,198 @@
+* Create a new app, and get your first API key.
+* Your API key will need the @publish@, @subscribe@, @presence@ and @history@ capabilities.
+* Install the Ably CLI: @npm install -g &#64;ably/cli@
+* Run @ably login@ in the terminal and follow the prompts to provide an access token to the Ably CLI.
+* Set the default app to use with @ably apps switch@ and select an API key to use for operations with @ably auth keys switch@.
+* Install the Ably Chat JavaScript SDK in a new project within your IDE. This will also install the Ably Pub/Sub SDK as it is a dependency:
+
+```[sh]
+npm install @ably/chat
+
+# React is temporarily required to be installed for the Chat SDK to function.
+npm install react
+```
+
+<aside data-type='note'>
+<p>The code examples in this guide include a demo API key. If you wish to interact with the Ably CLI, ensure that you replace them with your own API key.</p>
+</aside>
+
+h2(#step-1). Step 1: Connect to Ably
+
+Clients establish a connection with Ably when they instantiate an SDK. This enables them to send and receive messages in realtime across channels.
+
+* Run the following function to instantiate a realtime client with the Pub/Sub SDK and then pass that client into the Chat SDK constructor. At the minimum you need to provide an authentication mechanism and a @clientId@ to provide the client with an identity. Use an API key when authenticating for simplicity, but you should use token authentication in a production app:
+
+```[javascript]
+import * as Ably from 'ably';
+import { ChatClient, AllFeaturesEnabled, MessageEvent, Message } from '@ably/chat';
+
+async function getStarted() {
+
+  const realtimeClient = new Ably.Realtime({ key: '{{API_KEY}}', clientId: 'my-first-client' });
+
+  const chatClient = new ChatClient(realtimeClient);
+
+  const { off } = chatClient.connection.onStatusChange((change) => console.log(`Currently ${change.current}!`));
+
+};
+
+getStarted();
+```
+
+You can monitor the lifecycle of clients' connections, but for now just log a message to the console to know that the connection attempt was successful. You'll see a message saying @Currently connected!@ printed to your console.
+
+
+h2(#step-2). Step 2: Create a room and send a message
+
+Messages are how your clients interact with one another. Use rooms to separate and organize clients and messages into different topics, or 'chat rooms'. Rooms are the entry object into Chat, providing access to all of its features, such as messages, presence and reactions.
+
+* Add the following lines to your @getStarted()@ function to create an instance of a room, attach to the room instance, and then register a listener to subscribe to messages sent to the room. You then also send your first message. Afterwards, run it:
+
+```[javascript]
+const room = await chatClient.rooms.get('my-first-room', AllFeaturesEnabled);
+
+await room.attach();
+
+const { unsubscribe: messageUnsubscribe } = room.messages.subscribe((message: MessageEvent) => {
```

Comment:
Since we never unsubscribe maybe we can ignore the returned value? I know we should recommend to always unsubscribe/etc, but this is _getting started_, and we don't call `unsubscribe` anyway.

Reply:
Yes, that's fair. It's a balance balance between showing best practice and code that's never called. I think I agree with you on reflection so have removed them all. 3a0170cbe2b22520d7101a006a13af8251eed056